spring-integration-core/src/main/java/org/springframework/integration/history/MessageHistory.java

```diff
@@ -215,52 +217,52 @@ public String toString() {
 
 	@Override
 	public boolean add(Properties e) {
-		throw new UnsupportedOperationException("MessageHistory is immutable.");
+		throw new UnsupportedOperationException(MESSAGE_HISTORY_IS_IMMUTABLE);
```

Comment:
Can't we just have a whole `UnsupportedOperationException` as a singleton instance?
Or that way we are going to lose a method context for stack trace?

Reply:
I think it's ok to do this because the caller site will be in the stack trace.